.github/workflows/presubmit.yml

```diff
@@ -92,33 +26,18 @@ jobs:
         config-option: ["", "--enable-clang-parser"]
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/cache@v4
-        id: efficios_dep
-        env:
-          cache-name: cache-efficios_dep
-        with:
-          path: ~/efficios_dep/
-          key: ${{ runner.os }}-build-${{ env.cache-name }}
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE
-      - run: sudo gem install cast-to-yaml nokogiri babeltrace2 opencl_ruby_ffi metababel
-      - name: Load Babeltrace2
-        run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "PKG_CONFIG_PATH=$HOME/efficios_dep/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
-          echo "CPATH=$HOME/efficios_dep/include:$CPATH" >> $GITHUB_ENV
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
-          echo "LIBRARY_PATH=$HOME/efficios_dep/lib:$LIBRARY_PATH" >> $GITHUB_ENV
+      - uses: ./.github/actions/setup-thapi
       - name: Install dependencies for --enable-clang-parser
         if: matrix.config-option == '--enable-clang-parser'
         run: pip install git+https://github.com/TApplencourt/h2yaml
-      - run: mkdir -p build
-      - run: ./autogen.sh
-      - run: ../configure ${{ matrix.config-option }}
-        working-directory: build
-      - run: CC=gcc make -j
-        working-directory: build
-      - run: make check
-        working-directory: build
+      - name: Make check
+        run: |
+          ./autogen.sh
+          mkdir -p build
+          cd build
+          ../configure
+          CC=gcc make -j
+          make check
```

Comment:
I don't like these, this goes contradictory to the philosophy of github actions and increasing the diff for no good reason. I think that if we want to do that we should do so in a different PR.

Reply:
I can revert and do another PR for sure, but YAML is just a stupid format for any kind of programming or control flow. 

We should just use bash at maximum possible aka, limiting the number of `run` and magic `work-directory`, ` and stuff.

Reply:
I don't agree, we're loosing meta information by doing this. Let's chat about it.